mgmtsystem/virtualcenter.py

```diff
@@ -572,30 +655,31 @@ def clone_vm(self, source, destination, resourcepool=None, datastore=None, power
             folder = source_template.parent.parent.vmParent
         except AttributeError:
             folder = source_template.parent
-
         progress_callback("Picked folder `{}`".format(folder.name))
 
         task = source_template.CloneVM_Task(folder=folder, name=destination, spec=vm_clone_spec)
-
-        def _check():
+        start_time = time.time()
+        delta = 0
+        while task.info.state not in {'queued', 'running'} and delta < provision_timeout:
```

Comment:
The  code originally used wait_for, but the way self._update_obj() works I couldn't use wait_for anymore (at least not without making a global variable).